flink-python/src/main/java/org/apache/flink/client/python/PythonEnvUtils.java

```diff
@@ -137,6 +143,27 @@ static PythonEnvironment preparePythonEnvironment(
                             .collect(Collectors.toList());
             addToPythonPath(env, pythonFiles);
         }
+
+        // 5. set the archives directory as the working directory, then user could access the
+        // content of the archives via relative path
+        if (config.getOptional(PYTHON_ARCHIVES).isPresent()
+                && (config.getOptional(PYTHON_CLIENT_EXECUTABLE).isPresent()
+                        || !System.getenv(PYFLINK_CLIENT_EXECUTABLE).isEmpty())) {
+            env.archivesDirectory = String.join(File.separator, tmpDir, PYTHON_ARCHIVES_DIR);
+            List<Path> pythonArchives =
+                    Arrays.stream(config.get(PYTHON_ARCHIVES).split(FILE_DELIMITER))
+                            .map(Path::new)
+                            .collect(Collectors.toList());
+
+            // extract archives to archives directory
+            for (Path pythonArchive : pythonArchives) {
+                ZipUtils.extractZipFileWithPermissions(
+                        pythonArchive.getPath(),
+                        String.join(
+                                File.separator, env.archivesDirectory, pythonArchive.getName()));
```

Comment:
if python.archives is configured as venv.zip#myenv, the extracted directory name should be myenv